lib/connextclient/ConnextClient.ts

```diff
@@ -260,9 +261,27 @@ class ConnextClient extends SwapClient {
     return this._totalOutboundAmount.get(currency) || 0;
   }
 
-  public maxChannelInboundAmount = (_currency: string): number => {
-    // assume MAX_AMOUNT since Connext will re-collaterize accordingly
-    return MAX_AMOUNT;
+  public checkInboundCapacity = (inboundAmount: number, currency: string) => {
+    const inboundCapacity = this._maxChannelInboundAmount.get(currency) || 0;
+    if (inboundCapacity < inboundAmount) {
+      // we do not have enough inbound capacity to receive the specified inbound amount so we must request collateral
+      this.logger.debug(`collateral of ${inboundCapacity} for ${currency} is insufficient for order amount ${inboundAmount}`);
+      const capacityShortage = inboundAmount - inboundCapacity;
+      const capacityShortageUnits = this.unitConverter.amountToUnits({ currency, amount: capacityShortage });
+
+      // first check whether we already have a pending collateral request for this currency
+      // if not start a new request, and when it completes call channelBalance to refresh our inbound capacity
+      const requestCollateralPromise = this.requestCollateralPromises.get(currency) ?? this.sendRequest('/request-collateral', 'POST', {
+        assetId: this.tokenAddresses.get(currency),
+        amount: capacityShortageUnits * 1.05, // 5% buffer
+      }).then(() => {
+        this.logger.debug(`completed collateral request of ${capacityShortageUnits} ${currency} units`);
```

Comment:
The `/request-collateral` endpoint of rest-api-client returns immediately when the node has received the collateralization request. In theory, if the user spams the CLI for the order it could result in over-collateralizing the channel.

I'll run some manual tests to see what happens in this case. We might have to add an additional `inboundCapacity` check here to make sure that the collateralization has finished.

Reply:
After testing this manually it's not an issue. The `/request-collateral` request to rest-api-client will return once the channel is collateralized OR return immediately when the requested collateral amount exceeds the node's configured upper bound.

Reply:
Ok, that's perfect. Eventually we'd like to get an error when the amount exceeds the upper bound but for now this is OK.